prov/rxd/src/rxd_ep.c

```diff
@@ -147,14 +147,19 @@ struct fi_ops_ep rxd_ops_ep = {
 	.tx_size_left = fi_no_tx_size_left,
 };
 
-static int rxd_match_unexp_msg(struct dlist_entry *item, const void *arg)
+static int rxd_match_unexp(struct dlist_entry *item, const void *arg)
 {
-	const struct rxd_x_entry *rx_entry = arg;
+	struct rxd_x_entry *rx_entry = (struct rxd_x_entry *) arg;
 	struct rxd_pkt_entry *pkt_entry;
 
 	pkt_entry = container_of(item, struct rxd_pkt_entry, d_entry);
-	return (rx_entry->peer == FI_ADDR_UNSPEC ||
-		rx_entry->peer == pkt_entry->peer);
+
+	if (!rxd_match_addr(rx_entry->peer, pkt_entry->peer))
+		return 0;
+
+	return (rx_entry->cq_entry.flags & FI_TAGGED) ?
+		rxd_match_tag(rx_entry->cq_entry.tag, rx_entry->ignore,
```

Comment:
If you set non-tagged receives with ignore = ~0, you can collapse msg and tmsg paths into one.

Reply:
Doing so would let a normal message rx match to an unexpected tagged message. Is that allowed/expected?